### ISSUE TITLE
Components are missing when using IFC4

### ISSUE THREAD

**Issue by BasRongenSPIE** [Autodesk/revit-ifc#513]
Mail me for the files. I've also created a ticket (19541322).

When using IFC 3x2 everything exports. But using IFC 4 a lot is skipped:
![image](https://user-images.githubusercontent.com/70643737/192269735-6c936fe4-2d44-4fb8-bf15-842df1e3d6e8.png)

**Comment by AngelVelezSosa** [comment]
I don't have direct access to the tickets but should see it soon (if it isn't already on my JIRA board).  When you say IFC4, do you mean reference view?

**Comment by BasRongenSPIE** [comment]
Thank you for looking into it.

To answer your question: no, I’ve used IFC 4 Design Transfer.

Met vriendelijke groet / Yours faithfully,

Bas Rongen
BIM Manager

Mob:       +31 (0)6  22 58 70 53
***@***.******@***.***>
SPIE Nederland B.V.
Energies
Hoogspanningslijnen- en stations
Huifakkerstraat 15
NL-4815 PN Breda
www.spie-nl.com<http://www.spie-nl.com/>

**Comment by AngelVelezSosa** [comment]
OK, Design Transfer isn't "real".  There was a lot of promise for a Design Transfer View for IFC4 for a long time but now it will probably never materialize.  We have updated the UI to reflect that, to make it clear that this is "at your own risk" in the sense that it doesn't reflect a real MVD.  So it is possible that the data is there but the viewer doesn't show it.  What happens when you try to link it back into Revit?

**Comment by mathib** [comment]
@AngelVelezSosa some more context: the Revit file contains quite some complex geometry coming from Inventor. The resulting IFC2x3 export was pretty big (+500MB) for a rather small model (number of objects). We're trying to get better result by exporting to IFC4 using the Design Transfer MVD. 

Perhaps we better rely on IFC4 Reference View or do you expect the size to be similar to IFC2x3 TC1 Coordination View? Any recommendations (besides better transfer from Inventor to Revit)?

**Comment by AngelVelezSosa** [comment]
Well, I do recommend simplifying the Inventor geometry if you don't need the detail.  I believe Inventor does have shrinkwrapping - do you need your Inventor data in all of its glory in Revit?
What is your concern about the file size?  You can significantly reduce the file size by exporting as ifcZIP, which is a compressed format.  IFC4 RV has more compact meshes, but then you will get meshes which take more space than BReps.  So it is all a trade-off.

**Comment by BasRongenSPIE** [comment]
We're looking into reducing the 'glory' on the inventor side. Because we need more objects than just the [top-level-components ](https://knowledge.autodesk.com/support/inventor/learn-explore/caas/CloudHelp/cloudhelp/2022/ENU/Inventor-Help/files/GUID-9F0E010C-2295-4021-BB77-89C119ACDA1A-htm.html#:~:text=each%20top%20level%20component%20) we're forced to use the rfa export functionality for every object. this is very labour intensive, and will be replaced in the future, so for now we're stuck with this.

**Comment by AngelVelezSosa** [comment]
@mathib : I would agree that BReps are better than meshes.  It's important to figure out what the usage is for these BReps to make sure we set the right expectations for functionality.  The _primary_ use should be around referencing, but if there is an intention for shared editability, it is important to understand what that is.
